### DISCUSSION TITLE
Controller not updating value on change.

### DISCUSSION THREAD

**Discussion by nikhilkpn01** [react-hook-form/react-hook-form#11364] [Q&A]
I am using useFieldArray for a nested form. Were I fill the array with some default values. I need to change the value of the field based on index.
It works when I use "input" with 'register' but not working with 'Controller'.
Please find https://codesandbox.io/p/sandbox/usefieldarray-yz5yjs?file=%2Fsrc%2FApp.js%3A8%2C10
When I click on 'change' button both the fields should change. 
<img width="186" alt="Screenshot 2024-01-05 at 02 20 49" src="https://github.com/react-hook-form/react-hook-form/assets/126812960/288e6bc6-dc2f-4cb1-aa1f-171effc281c3">

Can someone help me to fix this issue?

**Comment by Moshyfawn** [comment]
I'm a bit confused about the use case here, so bear with me. I see you're registering a field array, but not using its API. By updating the `index` value in the name, you create 2 fields with **the same** name, which results in the last field with the name being registered as a field, overriding the behaviour of the one before it due to the order of operations. This wouldn't work.

**Reply by nikhilkpn01** [reply]
It's just an example. 
If you remove **NestedWithoutController** component it still won't work.
My use case is:
I have a deep nested form with more than 2 layer. Suppose I have,
`defaultValues: {
    students: [
        {
            name: '',
            subjects: [
                {
                    name: '',
                    mark: null
                }
            ]
        }
    ]
}`
And I keep only one form for filling single student details. After filling I can append another one and current form will be collapsed.
<img width="560" alt="Screenshot 2024-01-05 at 09 38 07" src="https://github.com/react-hook-form/react-hook-form/assets/126812960/316980ac-4b5c-4341-8e07-2171ecf72416">
Like the one used in **antd**. At a time I can edit/create a single student.
My requirement is when I pass the index  I want to open the form in edit mode so that I can update it.
